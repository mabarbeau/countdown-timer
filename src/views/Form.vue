<script>
import { defineComponent } from "vue";
import moment from "moment";

export default defineComponent({
  name: "From",
  data() {
    return {
      title: null,
      date: null
    };
  },
  computed: {
    utc() {
      return moment(this.date)
        .utc()
        .format();
    },
    isValid() {
      return moment(this.date).isValid();
    },
    params() {
      return this.date ? { date: this.utc } : undefined;
    },
    query() {
      return this.title ? { title: this.title } : undefined;
    }
  },
  methods: {
    onSubmit() {
      if (this.isValid) {
        this.$router.push({
          name: "Timer",
          params: this.params,
          query: this.query
        });
      }
    }
  }
});
</script>

<template>
  <form @submit.prevent="onSubmit">
    <label>
      Event Title
      <input v-model="title" type="text" />
    </label>
    <br />
    <label>
      Date
      <input
        v-model="date"
        type="datetime-local"
        min="1970-01-01T00:00"
        max="2970-01-01T00:00"
      />
    </label>
    <input type="submit" value="Start timer" />
  </form>
</template>
