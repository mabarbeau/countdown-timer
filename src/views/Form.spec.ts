import { mount } from "@vue/test-utils";
import Form from "@/views/Form.vue";

describe("Form.vue", () => {
  it("renders a form", () => {
    const wrapper = mount(Form);
    expect(wrapper.element).toMatchSnapshot();
  });
});
