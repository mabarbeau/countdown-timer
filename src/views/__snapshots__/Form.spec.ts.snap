// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form.vue renders a form 1`] = `
<form>
  <label>
     Event Title 
    <input
      type="text"
    />
  </label>
  <br />
  <label>
     Date 
    <input
      max="2970-01-01T00:00"
      min="1970-01-01T00:00"
      type="datetime-local"
    />
  </label>
  <input
    type="submit"
    value="Start timer"
  />
</form>
`;
